diff --git a/modules/juce_graphics/native/juce_Fonts_android.cpp b/modules/juce_graphics/native/juce_Fonts_android.cpp
index 469f9f3..e5c02df 100644
--- a/modules/juce_graphics/native/juce_Fonts_android.cpp
+++ b/modules/juce_graphics/native/juce_Fonts_android.cpp
@@ -273,8 +273,10 @@ public:
 
     Typeface::Ptr createSystemFallback (const String& text, const String& language) const override
     {
+       #if __ANDROID_API__ >= 29
         if (__builtin_available (android 29, *))
             return matchWithAFontMatcher (text, language);
+       #endif
 
         // The font-fallback API is only available on Android API level 29+
         jassertfalse;
@@ -292,12 +294,15 @@ public:
 
     static Typeface::Ptr findSystemTypeface()
     {
+       #if __ANDROID_API__ >= 29
         if (__builtin_available (android 29, *))
             return findSystemTypefaceWithMatcher();
+       #endif
 
         return from (FontOptions{}.withName ("Roboto"));
     }
 
+   #if __ANDROID_API__ >= 29
     static JUCE_INTRODUCED_IN_29 Typeface::Ptr findGenericTypefaceWithMatcher (const Font& font)
     {
         using AFontMatcherPtr = std::unique_ptr<AFontMatcher, FunctionPointerDestructor<AFontMatcher_destroy>>;
@@ -347,6 +350,7 @@ private:
 
         return fromMatchedFont (matched.get());
     }
+   #endif
 
     static bool shouldStoreAndroidFont (hb_face_t* face)
     {
@@ -805,6 +811,7 @@ Typeface::Ptr Font::Native::getDefaultPlatformTypefaceForFont (const Font& font)
 {
     const auto faceName = font.getTypefaceName();
 
+   #if __ANDROID_API__ >= 29
     const auto idealFace = std::invoke ([&]() -> Typeface::Ptr
     {
         if (__builtin_available (android 29, *))
@@ -827,6 +834,9 @@ Typeface::Ptr Font::Native::getDefaultPlatformTypefaceForFont (const Font& font)
 
     if (idealFace != nullptr)
         return idealFace;
+   #else
+    ignoreUnused (faceName);
+   #endif
 
     Font f (font);
     f.setTypefaceName (std::invoke ([&]() -> String
